File: src/main/java/by/miaskor/java/creational/factory_method/MainApp.java
package by.miaskor.java.creational.factory_method;

import by.miaskor.java.creational.factory_method.factory.BoatFactory;
import by.miaskor.java.creational.factory_method.factory.TruckFactory;

public class MainApp {

  public static void main(String[] args) {
    System.out.println(new BoatFactory().make());
    System.out.println(new TruckFactory().make());
  }

}

File: src/main/kotlin/by/miaskor/kotlin/creational/factory_method/MainApp.kt
package by.miaskor.kotlin.creational.factory_method

import by.miaskor.kotlin.creational.factory_method.factory.BoatFactory
import by.miaskor.kotlin.creational.factory_method.factory.TransportFactory
import by.miaskor.kotlin.creational.factory_method.factory.TruckFactory

fun main() {
  val boatFactory: TransportFactory = BoatFactory()
  val truckFactory: TransportFactory = TruckFactory()

  println(boatFactory.make())
  println(truckFactory.make())
}
